core/field.js

```diff
@@ -235,26 +238,64 @@ Blockly.Field.prototype.initView = function() {
   if (!this.visible_) {
     this.fieldGroup_.style.display = 'none';
   }
+  this.createBorderRect_();
+  this.createTextElement_();
+};
+
+/**
+ * Create a field border rect element. Not to be overridden by subclasses.
+ * Instead modify the result of the function inside initView, or create a
+ * separate function to call.
+ * @protected
+ */
+Blockly.Field.prototype.createBorderRect_ = function() {
   this.borderRect_ = Blockly.utils.createSvgElement('rect',
       {
         'rx': 4,
         'ry': 4,
         'x': -Blockly.BlockSvg.SEP_SPACE_X / 2,
         'y': 0,
-        'height': 16
+        'height': 16,
+        'width': this.size_.width + Blockly.BlockSvg.SEP_SPACE_X
       }, this.fieldGroup_);
-  this.textElement_ = Blockly.utils.createSvgElement('text',
-      {'class': 'blocklyText', 'y': this.size_.height - 12.5},
-      this.fieldGroup_);
-  var textNode = document.createTextNode('');
-  this.textElement_.appendChild(textNode);
+};
 
-  this.updateEditable();
+/**
+ * Create a field text element. Not to be overridden by subclasses. Instead
+ * modify the result of the function inside initView, or create a separate
+ * function to call.
+ * @protected
+ */
+Blockly.Field.prototype.createTextElement_ = function() {
+  this.textElement_ = Blockly.utils.createSvgElement('text',
+      {
+        'class': 'blocklyText',
+        'y': this.size_.height - 12.5
+      }, this.fieldGroup_);
+  this.textContent_ = document.createTextNode('');
+  this.textElement_.appendChild(this.textContent_);
+};
 
-  this.clickTarget_ = this.getSvgRoot();
+/**
+ * Bind events to the field. Only used for events that absolutely apply to
+ * all fields, such as tooltip events. Other events can be applied on a
+ * class-by-class basis using doEventBinding_.
+ * @private
+ */
+Blockly.Field.prototype.fieldBindEvents_ = function() {
+  Blockly.Tooltip.bindMouseEvents(this.getClickTarget_());
   this.mouseDownWrapper_ =
       Blockly.bindEventWithChecks_(
-          this.clickTarget_, 'mousedown', this, this.onMouseDown_);
+          this.getClickTarget_(), 'mousedown', this, this.onMouseDown_);
+  this.doEventBinding_();
+};
+
+/**
+ * Bind events to the field. Can be overridden by subclasses.
+ * No-op by default.
+ * @protected
+ */
+Blockly.Field.prototype.doEventBinding_ = function() {
```

Comment:
Thinking about this again, this doesn't really need to be two functions. It'd probably be just as easy for developers to override a protected bindEvents_ function and call the parent (which would do the tooltip and click stuff). That would also make it easier for devs to remove the tooltip and click events if they interfere with the input they're trying to recieve.

Does changing this to one protected function sound ok?

Reply:
Fixed